persistence/src/main/java/com/epam/esm/dao/util/GiftCertificateExtractorUtil.java

```diff
@@ -0,0 +1,93 @@
+package com.epam.esm.dao.util;
+
+import com.epam.esm.model.GiftCertificate;
+import com.epam.esm.model.Tag;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class GiftCertificateExtractorUtil {
+
+    private static final String CERTIFICATE_ID_COLUMN = "ID";
+    private static final String CERTIFICATE_NAME_COLUMN = "Name";
+    private static final String CERTIFICATE_DESCRIPTION_COLUMN = "Description";
+    private static final String CERTIFICATE_PRICE_COLUMN = "Price";
+    private static final String CERTIFICATE_CREATE_DATE_COLUMN = "CreateDate";
+    private static final String CERTIFICATE_LAST_UPDATE_DATE_COLUMN = "LastUpdateDate";
+    private static final String CERTIFICATE_DURATION_COLUMN = "Duration";
+
+    private static final String TAG_ID_COLUMN = "TAGS.ID";
+    private static final String TAG_NAME_COLUMN = "TAGS.Name";
+
+    private GiftCertificateExtractorUtil(){
+    }
+
+    public static List<GiftCertificate> extractGiftCertificates(ResultSet rs) throws SQLException {
+        Map<Integer, GiftCertificate> map = new LinkedHashMap<>();
+        while (rs.next()) {
+            setCertificateToMap(rs, map);
+        }
+
+        return new ArrayList<>(map.values());
+    }
+
+    public static GiftCertificate extractGiftCertificate(ResultSet rs) throws SQLException {
+        Map<Integer, GiftCertificate> map = new LinkedHashMap<>();
```

Comment:
Fix, refactor to HashMap

Reply:
> Why LinkedHashMap, but not HashMap

Because if using HashMap, we will receive certificates in the order in which they are stored (depending on the hash code). But when we need to receive certificates sorted by some param, this is unacceptable.